scripts/render_notebook_viz.py

```
@@ -0,0 +1,290 @@
+    def set_notebook(self, notebook_path: Path) -> None:
+        self._notebook = notebook_path
+        self._counter = 0
+
+    def set_original(self, visualize_func: Callable[..., Any]) -> None:
+        self._orig_visualize = visualize_func
+
+    def _slugify(self, value: str) -> str:
+        slug = re.sub(r"[^a-zA-Z0-9_-]+", "_", value).strip("_")
+        return slug or "graph"
+
+    def _next_filepath(self, graph: Any, kwargs: dict[str, Any]) -> Path:
```

Ruff (0.14.14)

[warning] Unused method argument: `kwargs`

(ARG002)